File: trunk/net.tenteCsharp.templatesProject/src_code/lightManagement/CentralGUI.java


cclass lightManagement.LightManagement; 

public cclass CentralGUI extends TypeComponent{ 

	public LightNotifyPort lightNotifyPort;
	
	
	public CentralGUI (String id){
		super(id);
	}
	
	public LightNotifyPort getLightNotifyPort(){
		return lightNotifyPort;
	}
	
	public void setLightNotifyPort(CentralGUI.LightNotifyPort value){
		this.lightNotifyPort=value;
	}
	

	
	
	public cclass LightNotifyPort extends TypePort{

		public ArrayList portsIGeneralLightNotify;

		public LightNotifyPort (){
			super();
			portsIGeneralLightNotify=new ArrayList();

		}


			
		public void connectPort(IGeneralLightNotify port){
			portsIGeneralLightNotify.add(port);
		}
	}

}

File: net.tenteCsharp/src-gen/lightManagement/CentralGUI.cs
using System;
using System.Collections;
using System.Text;

namespace SmartHome
{
	public partial class CentralGUI : TypeComponent
	{
		public LightNotifyPort lightNotifyPort;
	
		public CentralGUI (String id) 
			: base(id)
		{
		}
		public LightNotifyPort getLightNotifyPort()
		{
			return lightNotifyPort;
		}
	
		public void setLightNotifyPort(CentralGUI.LightNotifyPort value)
		{
			this.lightNotifyPort=value;
		}
	

	
		
		public class LightNotifyPort : TypePort
		{
 		public ArrayList portsIGeneralLightNotify = new ArrayList();

			public LightNotifyPort()
				: base()
			{
				portsIGeneralLightNotify=new ArrayList();

			}


			public ArrayList getPortsIGeneralLightNotify()
			{
	        	return portsIGeneralLightNotify;
	        }
				
			public void connectPort(IGeneralLightNotify port)
			{
				portsIGeneralLightNotify.Add(port);
			}
	
		}
	}
}